### PR TITLE
Add a tool 'uplift_beta' to set a ni when a bug is fixed in nightly and is affecting beta

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: auto_nag/scripts/uplift_beta.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            return
+
+        self.beta = self.versions["beta"]
+        self.status_central = utils.get_flag(
+            self.versions["central"], "status", "central"
+        )
+        self.status_beta = utils.get_flag(self.beta, "status", "beta")
+
+    def description(self):
+        return "Bugs fixed in nightly but still affect other supported channels"
```
Comment (marco-c) [inline auto_nag/scripts/uplift_beta.py]:
```suggestion
        return "Bugs fixed in nightly but still affecting other supported channels"
```

File: auto_nag/scripts/uplift_beta.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    def has_assignee(self):
+        return True
```
Comment (marco-c) [inline auto_nag/scripts/uplift_beta.py]: ?

File: auto_nag/scripts/uplift_beta.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            "id": bugid,
+            "mail": assignee,
+            "nickname": nickname,
+            "summary": self.get_summary(bug),
+            "regressions": bug["regressions"],
+        }
+        return bug
+
+    def filter_bugs(self, bugs):
+        # Get the bugs where status_central has been set few days ago
```
Comment (marco-c) [inline auto_nag/scripts/uplift_beta.py]: Can't we rely on status moving to FIXED, and so bake this directly in the Bugzilla query?

File: auto_nag/scripts/uplift_beta.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        bugids = list(bugs.keys())
+        data = set()
+        Bugzilla(
+            bugids=bugids, historyhandler=history_handler, historydata=data
+        ).get_data().wait()
+
+        return {bugid: info for bugid, info in bugs.items() if bugid in data}
+
+    def filter_by_regr(self, bugs):
+        # Filter the bugs which don't have any regression or where the regressions are all closed
```
Comment (marco-c) [inline auto_nag/scripts/uplift_beta.py]: Maybe when there are regressions that were fixed, we should note that in the comment and remind them to uplift all patches

File: auto_nag/scripts/uplift_beta.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+        fixed_bugs = set()
+        Bugzilla(
+            bugids=list(bugids),
+            include_fields=["id", "status"],
+            bughandler=bug_handler,
+            bugdata=fixed_bugs,
+        ).get_data().wait()
+
+        new_bugs = {}
```
Comment (marco-c) [inline auto_nag/scripts/uplift_beta.py]:
```suggestion
        bugs_without_regressions = {}
```

File: templates/uplift_beta_needinfo.txt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+The patch landed in nightly and beta is affected.
+:{{ nickname }}, is that bug important enough to require an uplift?
```
Comment (marco-c) [inline templates/uplift_beta_needinfo.txt]:
```suggestion
:{{ nickname }}, is this bug important enough to require an uplift?
```

File: auto_nag/scripts/uplift_beta.py (hunk 2 above)
Comment (calixteman) [reply auto_nag/scripts/uplift_beta.py]: Needed to have assignee info

File: auto_nag/scripts/uplift_beta.py (hunk 4 above)
Comment (calixteman) [reply auto_nag/scripts/uplift_beta.py]: I'm not sure about the wording

File: auto_nag/scripts/uplift_beta.py (hunk 2 above)
Comment (marco-c) [reply auto_nag/scripts/uplift_beta.py]: Maybe we should rename the function, as it's not super clear

File: auto_nag/scripts/uplift_beta.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+            "include_fields": fields,
+            "bug_type": "defect",
+            "resolution": ["---", "FIXED"],
+            "f1": self.status_central,
+            "o1": "anyexact",
+            "v1": ",".join(["fixed", "verified"]),
+            "f2": self.status_beta,
+            "o2": "anyexact",
+            "v2": "affected",
+            # Changed before 2 days ago and not changed after 2 days ago
```
Comment (jcristau) [inline auto_nag/scripts/uplift_beta.py]: Does "last resolved" not give you what you need here?

Comment (calixteman) [reply auto_nag/scripts/uplift_beta.py]: I'd say that `last_resolved` and this way to check are pretty similar but different.
AFAIK, `cf_last_resolved` is the last time that `resolution` is no more `---` or something like that and here we check the last change on the status flag for nightly.
Anyway, I don't care so much, finally it's up to you, so what's the best option according to your needs ?

**Pull request merged**